File: Arrays/MajorityElementII.cpp
#include <vector>
#include <unordered_map>


class Solution {
public:
    std::vector<int> majorityElement(std::vector<int>& nums) {
        std::unordered_map<int, int> elementFrequencyMap;


        for (int num : nums) {
            elementFrequencyMap[num]++;
        }


        std::vector<int> majorityElements;


        for (auto& pair : elementFrequencyMap) {
            if (pair.second > nums.size() / 3) {
                majorityElements.push_back(pair.first);
            }
        }


        return majorityElements;
    }
};

File: Arrays/MajorityElementII.js
var majorityElement = function(nums) {
    const elementFrequencyMap = new Map();


    for (const num of nums) {
        elementFrequencyMap.set(num, (elementFrequencyMap.get(num) || 0) + 1);
    }


    const majorityElements = [];


    for (const [num, count] of elementFrequencyMap) {
        if (count > Math.floor(nums.length / 3)) {
            majorityElements.push(num);
        }
    }


    return majorityElements;
};
